datafusion/physical-expr/src/expressions/is_null.rs

```diff
@@ -117,6 +117,21 @@ pub(crate) fn compute_is_null(array: ArrayRef) -> Result<BooleanArray> {
     }
 }
 
+/// workaround <https://github.com/apache/arrow-rs/issues/6017>,
+/// this can be replaced with a direct call to `arrow::compute::is_not_null` once it's fixed.
+pub(crate) fn compute_is_not_null(array: ArrayRef) -> Result<BooleanArray> {
```

Comment:
this code looks good to me -- I think it also ends up supporting `UnionArray` in `IS NOT NULL`

cc @samuelcolvin who added this code in https://github.com/apache/datafusion/pull/11321